CefSharp.OffScreen/ChromiumWebBrowser.cs

```diff
@@ -320,24 +339,15 @@ public void Dispose()
         }
 
         /// <summary>
-        /// Releases unmanaged and - optionally - managed resources.
+        /// Releases unmanaged and - optionally - managed resources for the <see cref="ChromiumWebBrowser"/>
         /// </summary>
-        /// <param name="disposing"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
+        /// <param name="disposing"><see langword="true" /> to release both managed and unmanaged resources; <see langword="false" /> to release only unmanaged resources.</param>
         protected virtual void Dispose(bool disposing)
         {
-            // Don't reference event listeners any longer:
-            AddressChanged = null;
-            BrowserInitialized = null;
-            ConsoleMessage = null;
-            FrameLoadEnd = null;
-            FrameLoadStart = null;
-            LoadError = null;
-            LoadingStateChanged = null;
-            Paint = null;
-            StatusMessage = null;
-            TitleChanged = null;
-
-            Cef.RemoveDisposable(this);
+            if (Interlocked.CompareExchange(ref disposeSignaled, 1, 0) != 0)
```

Comment:
I'm thinking we need a comment for this line in the three instances, as it does more than just check if the count is 0 or 1.

Thoughts?

Reply:
I feel it explains itself, the method is called CompareExchange and if the user hovers over it they can see exactly what it does (compare and if equal, exchange + return original value).

Reply:
For anyone unfamiliar with `CompareExchange` it's kinda cryptic. It maybe super clear now, if you come back in 6-12 months it might not be so clear

At a cursory glance even I though, can this be changed to use the `IsDisposed` property.

I'll add a comment later.

Reply:
Yeah, you're right.

Sure